### ISSUE TITLE
feature request: expand code style settings for greater control of indents and braces

### ISSUE THREAD

**Issue by patanne** [intellij-rust/intellij-rust#4576]
## Environment

* **IntelliJ Rust plugin version:** 0.2.108.*
* **IDE name and version:** CLion 2019.2.4

## Problem Description
I want Allman-style braces. I find K&R to be completely unreadable; so much so I will not use Go ever and have sidelined Rust to being "of interest" only, albeit extreme interest. 

## Current State
The current plugin lacks the code style options available for C/C++ in CLion or C# in Rider, options that provide fine-grained control over the alignment and placement of braces.

This missing ability also leaves the "Reformat Code" option of limited use to reset spaces to tab-based indents, or anything else, because it also undoes all the manual effort to achieve Allman.

## Simple Solution
In IntelliJ, for Kotlin, there is a code style option to place left braces on a new line. This is a blunt option but would actually be an excellent, quick solution.

![braces_placement](https://user-images.githubusercontent.com/17474951/67762484-ef751700-fa1b-11e9-818f-f26a23f455f1.png)

**Comment by ta32** [comment]
yes please give us the option to put braces on new lines. I don't think we should force a code style
